taipy/core/scenario/_scenario_manager.py

```diff
@@ -261,6 +261,17 @@ def _get_all_by_cycle(cls, cycle: Cycle) -> List[Scenario]:
             fil.update({"cycle": cycle.id})
         return cls._get_all_by(filters)
 
+    @classmethod
```

Comment:
Can you help add some tests for this function in `tests/core/scenario/test_scenario_manager.py`

Reply:
Sure , but  I am a bit confused by the referred  function , is it the   `_get_all_by()` or the` _get_all_by_cycle()` ?

Reply:
not sure if I understand your question correctly, but if it's regarding the naming of the function, I think `_get_scenarios_by_time` is perfect for the job 😃  some more context:
- `_get_all_by `: is a general get scenarios by some params function 
- `_get_all_by _cycle`: is a specific get function aims to quickly and simply get scenarios based on the provided cycle
Let me know if that answers your question!

Reply:
Please have a look at this update [Added test on _get_all_by_cycle()](https://github.com/Avaiga/taipy/pull/646/commits/aed2cc4207427f95991ecac8c3c4bb68bafe5a07)

Reply:
oh thanks! ah haha I actually meant a test for `_get_scenarios_by_time` in `tests/core/scenario/test_scenario_manager.py`. Sorry for not being clear about it 😅